packages/template-retail-react-app/app/constants.js

```diff
@@ -201,3 +201,19 @@ export const STORE_LOCATOR_DISTANCE = 100
 export const STORE_LOCATOR_NUM_STORES_PER_LOAD = 10
 export const STORE_LOCATOR_DISTANCE_UNIT = 'km'
 export const STORE_LOCATOR_IS_ENABLED = true
+
+// Constants for shopper context
+export const SHOPPER_CONTEXT_SEARCH_PARAMS = {
+    SOURCE_CODE: 'sourceCode',
+    EFFECTIVE_DATE_TIME: 'effectiveDateTime',
+    CUSTOMER_GROUP_IDS: 'customerGroupIds'
+}
+export const SHOPPER_CONTEXT_CUSTOM_QUALIFIERS_SEARCH_PARAMS = {
+    DEVICE_TYPE: 'deviceType',
+    IP_ADDRESS: 'ipAddress',
+    OPERATING_SYSTEM: 'operatingSystem'
+}
+export const SHOPPER_CONTEXT_ASSIGNMENT_QUALIFIERS_SEARCH_PARAMS = {
+    STORE: 'store'
+}
+export const SHOPPER_CONTEXT_ARRAY_FIELDS = [SHOPPER_CONTEXT_SEARCH_PARAMS.CUSTOMER_GROUP_IDS]
```

Comment:
here is where devs can define the name of the search param they want to set for each shopper context key

Reply:
TODO: test B2C Commerce version 24.7 with `geolocation` and `clientIp`
https://developer.salesforce.com/docs/commerce/commerce-api/references/shopper-context?meta=updateShopperContext

Reply:
current BM is alraedy 24.9. Logged into your BM and take a look at bottom right corner
![image](https://github.com/user-attachments/assets/71d09add-982b-4acc-8e30-7f10c7bbc9d1)

Reply:
Do you mean this is the "allowed" params that be used for context? 🤔

Reply:
Yes this is where the customer lists the query params whose values should be extracted and forwarded to the shopper context API.